UPGRADING.md

```diff
@@ -1,5 +1,13 @@
 # Upgrade guidelines
 
+## 0.46.0 to 0.46.1
+
+[#421 extra operators for container logs are inserted after all default data
+processes](https://github.com/signalfx/splunk-otel-collector-chart/pull/421)
+
+If you are referencing fields in log body (ie, `$$.pod_name`), change them
+corresponding attributes/resource (ie, `$$resource["k8s.pod.name"]`).
```

Comment:
Let's mention `logsCollection.containers.extraOperators` config
```suggestion
If you have any custom log processing operators defined in `logsCollection.containers.extraOperators`
configuration, change any references to log body fields (ie, `$$.pod_name`) in the operators to
corresponding resource attributes (ie, `$$resource["k8s.pod.name"]`).
```

Another question, do body fields `$$.pod_name` really need to be replaces with resource attributes `$$resource["k8s.pod.name"]`, not log attributes `$$attributes["k8s.pod.name"]`?

Reply:
```
      # Move out attributes to Attributes
      - type: metadata
        resource:
          k8s.pod.uid: 'EXPR($$.uid)'
          k8s.container.restart_count: 'EXPR($$.restart_count)'
          k8s.container.name: 'EXPR($$.container_name)'
          k8s.namespace.name: 'EXPR($$.namespace)'
          k8s.pod.name: 'EXPR($$.pod_name)'
          com.splunk.sourcetype: 'EXPR("kube:container:"+$$.container_name)'
        attributes:
          log.iostream: 'EXPR($$.stream)'
```

This is what we have. so it is added to resource.

Reply:
we can put them all under attributes. it will be the same. I don't think there will be any improvements? can i ask why you want to use log attributes?

Reply:
It's not about what we _want_ to use. We have this operator `clean-up-log-record` that translates log representation from one format to another. We charge expected input for `logsCollection.containers.extraOperators` and the custom operators will work with different format. We need to provide users explicit guidance on what to change in their `extraOperators` config to keep the same behavior.

You are saying that they need to replace `$$.pod_name` to `$$resource["k8s.pod.name"]`, but I don't think it's correct. As far as I see `clean-up-log-record` moves everything from `log` attribute to the root of the entry so I believe instructions must be different something like replacing `$$.log` with `$$`. Please correct me if I'm wrong.

Reply:
$$.pod_name to $$resource["k8s.pod.name"] 
^ this was just one example. 
all parsed metadata is added as either resource/attributes. so users should grep it from resource/attributes instead of trying to get it from log body like `$$.pod_name`. 
I agree that I need to add another line for `$$.log`.

Reply:
@dmitryax Could you help me understand what seems incorrect to you?

Reply:
> $$.pod_name to $$resource["k8s.pod.name"]
> ^ this was just one example.
> all parsed metadata is added as either resource/attributes. so users should grep it from resource/attributes instead of trying to get it from log body like $$.pod_name.

This is correct, but user has to do this even before this change. `$$.pod_name` gives nothing within `extraOperators` before and after this change, right? If so, why should it be added to the upgrade guidelines?

Reply:
before the upgrade, it would certainly yield pod name. since it comes before `clean-up-log-record`, body would have that field.